File: python/performance_test.py
import unittest
import time
from main import countWellFormedParenthesis

class PerformanceTest(unittest.TestCase):
    def test_performance(self):
        n = 15
        start = time.perf_counter()
        result = countWellFormedParenthesis(n)
        end = time.perf_counter()

        duration_ms = (end - start) * 1000  # в миллисекундах

        print(f"Result: {result}")
        print(f"Execution time: {duration_ms:.3f} ms")

        self.assertGreater(result,0)
        self.assertLess(duration_ms,200)

File: src/test/java/gts/PerformanceTest.java
package gts;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

class PerformanceTest {
    @Test
    void testPerformance() {
        long start = System.nanoTime();
        int result = Count.countWellFormatedParenthesis(15);
        long end = System.nanoTime();

        double durationMs = (end - start) / 1_000_000.0;

        System.out.println("Result: " + result);
        System.out.printf("Execution time (ms): %.6f%n", durationMs);

        assertTrue(end - start < 1_000_000); // < 1 миллисекунды (в наносекундах)
    }

}